### PR TITLE
basic-effect: Implement basic Effect class

### PR DESCRIPTION
**Pull request by elias-gvin**
1. Implement basic Effect class
2. Update StealEffect
3. Update class diagramm
4. Update utils module

### CODE REVIEW

File: effects/effect.py [hunk 1]
```diff
@@ -17,8 +15,75 @@ class EffectStatus(Enum):
...
+        self._status = EffectStatus.CREATED
+        self._activation_round = 0
+
+    def activate(self) -> None:
+        if self._status == EffectStatus.CREATED:
+            is_activated = self._activate_impl()
+
+            if (is_activated):
+                self._status = EffectStatus.ACTIVATED
+                self._activation_round = self.game.round_number
```
Comment (elias-gvin) [inline effects/effect.py:39]: Possibly, warning should be added. Effect should be always activated succesfully

File: user_interaction.py [hunk 2]
```diff
@@ -23,7 +26,7 @@ def show_active_instant(text: str) -> None:
     print(text)
 
 
-def read_index(text: str) -> int:
```
Comment (asternotus) [inline user_interaction.py:26]: Why read_index is removed? It was your idea about "read_index" method name.

File: utils.py [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-
 from typing import List
 
+from citizens.citizen import Citizen
+
 
-def is_citizen_in_range(target_number: int, citizens: List[Citizen]) -> bool:
-    return True
+def is_citizen_in_range(target_index: int, citizens: List[Citizen]) -> bool:
+    return target_index >= 0 and target_index < citizens.count()
```
Comment (asternotus) [inline utils.py]: Why do you not use len(citizens) to get the list size?

Comment (elias-gvin) [reply utils.py]: Agree. Done.

File: user_interaction.py (hunk 2 above)
Comment (elias-gvin) [reply user_interaction.py:26]: I've changed my mind. Word "index" is misleading in this context, in my oppinion.
Index means some number by which we can navigate through list or some other container.
But user don't write index of citizen but rater - (index+1)
So, I've decided to change method name from "read_index" to "read_number" to emphasize this difference.

File: effects/effect.py (hunk 1 above)
Comment (elias-gvin) [reply effects/effect.py:39]: Done.

File: user_interaction.py (hunk 2 above)
Comment (elias-gvin) [reply user_interaction.py:26]: One more comment. Mb you are right and we should keep 'read_index' name.
But we should get rid of "target_number - 1" logic first.
I suggest to count effects in card_list from '0'. But '0' will mean - "Do nothing".
But there are still problem with citizen list. What should be '0' in this case?
Should be discussed

File: utils.py (hunk 3 above)
Comment (asternotus) [reply utils.py]: Approved.

File: user_interaction.py (hunk 2 above)
Comment (asternotus) [reply user_interaction.py:26]: Set "read_number" name, please!

**Review by asternotus — APPROVED** [review]
Done. Rebase it, please!

**Pull request merged**